docs/dev/table/tuning/streaming_aggregation_optimization.md

```diff
@@ -0,0 +1,271 @@
+
+The following examples show how to enable the option.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+// instantiate table environment
+TableEnvironment tEnv = ...
+
+tEnv.getConfig()        // access high-level configuration
+  .getConfiguration()   // set low-level key-value options
+  .setString("table.exec.mini-batch.enabled", "true")  // enable mini-batch optimization
+  .setString("table.exec.mini-batch.allow-latency", "5 s") // use 5 seconds to buffer input records
+  .setString("table.exec.mini-batch.size", "5000"); // the maximum number of records can be buffered by each aggregate operator tasks
+{% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+// instantiate table environment
+val tEnv: TableEnvironment = ...
+
+tEnv.getConfig         // access high-level configuration
+  .getConfiguration    // set low-level key-value options
+  .setString("table.exec.mini-batch.enabled", "true") // enable mini-batch optimization
+  .setString("table.exec.mini-batch.allow-latency", "5 s") // use 5 seconds to buffer input records
+  .setString("table.exec.mini-batch.size", "5000") // the maximum number of records can be buffered by each aggregate operator tasks
+{% endhighlight %}
+</div>
+
+<div data-lang="python" markdown="1">
+{% highlight python %}
+# instantiate table environment
+t_env = ...
+
+t_env.get_config()        # access high-level configuration
+  .get_configuration()    # set low-level key-value options
+  .set_string("table.exec.mini-batch.enabled", "true") # enable mini-batch optimization
+  .set_string("table.exec.mini-batch.allow-latency", "5 s") # use 5 seconds to buffer input records
+  .set_string("table.exec.mini-batch.size", "5000"); # the maximum number of records can be buffered by each aggregate operator tasks
+{% endhighlight %}
+</div>
+</div>
+
+## Local-Global Aggregation
+
+Local-Global is proposed to solve data skew problem by dividing an group aggregation into two stages, that is doing local aggregation in upstream firstly, and followed by global aggregation in downstream, which is similar to Combine + Reduce pattern in MapReduce. For example, considering the following SQL:
+
+{% highlight sql %}
+SELECT color, sum(id)
+FROM T
+GROUP BY color
+{% endhighlight %}
+
+It is possible that the records in the data stream are skewed, thus some instances of aggregation operator have to process much more records than others, which leads to hotspot.
+The local aggregation can help to accumulate a certain amount of inputs which have the same key into a single accumulator. The global aggregate will only receive the reduced accumulators instead of large number of raw inputs.
```

Comment:
"The global **aggregate** will" -> "The global **aggregation** will".
keep same as mentioned above.